### PR TITLE
New GetMappingsView API and CreateIndexMappings API improvements

### LINKED ISSUE #44: [FEATURE] Mappings View and apply flow APIs
**Is your feature request related to a problem?**

Frontend needs capability to display a view of field mappings for a given **_index_name_** and _**_rule_topic_.**_ That view would have list of fields from index with corresponding aliases from template alias mappings. User would be able to edit given View of mappings and apply them on index. Edit would allow picking different alias for field from a list of returned aliases. 
**CreateIndexMappings** API currently applies predefined alias mappings designated by **_rule_topic_** parameter. We need to extend capability of API to accept alias mappings from user too.

**What solution would you like?**
Implement new API/Action GetMappingsViewAction. 

For a given **_index_name_** parameter, returns list of index fields and proposed aliases which are defined in template alias mappings file, designated by **rule_topic** parameter.
Example:

**REQUEST:**
``` 
GET _plugins/_security_analytics/mappings/view?index_name=sample_index&rule_topic=netflow
```
**RESPONSE:**
```
{
  "properties": {
    "source.ip": {
      "type": "alias",
      "path": "netflow.source_ipv4_address"
    },
    "source.port": {
      "type": "alias",
      "path": "netflow.source_transport_port"
    },
    "destination.ip": {
      "type": "alias",
      "path": "netflow.destination_ipv4_address"
    },
    "destination.port": {
      "type": "alias",
      "path": "netflow.destination_transport_port"
    },
    "http.request.method": {
      "type": "alias",
      "path": "netflow.http_request_method"
    },
    "http.response.status_code": {
      "type": "alias",
      "path": "netflow.http_status_code"
    }
  },
  "unmapped_index_fields": [
      "src_client_ip",
      "user_source_ip"
  ],
  "unmapped_field_aliases": [
      "source_ip",
      "client_ip"
  }
}
```
Response will include following:
| property | description |
| --- | --- |
| **_properties_** | applicable alias mappings to a given index. |
| **_unmapped_index_fields_** | list of fields in given index for which there are no found alias mappings |
| **_unmapped_field_aliases_** | list of aliases which path didn't match any field in given index |
### 
**_CreateIndexMappings API changes_**
###
Currently this API accepts only **_index_name_**, **_rule_topic_** and **_partial_** parameters. New parameter **_alias_mappings_** will contain json alias mappings provided by user. These mappings will be applied to index, instead of **_rule_topic_** ones. Example:

**REQUEST:**
```
POST _plugins/_security_analytics/mappings
{
   "index_name": "sample_index",
   "alias_mappings": {
        "properties": {
           "source.ip": {
              "type": "alias",
              "path": "netflow.source_ipv4_address"
           },
           "source.port": {
              "type": "alias",
              "path": "netflow.source_transport_port"
           }
       }
   }
}
```
**RESPONSE:**
```
{
   "acknowledged": true
} 
```

**Do you have any additional context?**

### PR DESCRIPTION
**Pull request by petardz**
### Description
Added new API GetMappingsView which returns mappings view with given index name and rule topic parameters.

Improved CreateIndexMappings API to accept optional parameter alias_mappings which is user-supplied alias mappings. This parameter will have higher priority over rule_topic parameter.
 
### Issues Resolved
#44 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/mapper/MapperTopicStore.java [hunk 1]
```diff
@@ -5,35 +5,33 @@
...
 import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.settings.SettingsException;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.json.JsonXContent;
 
-public class MapperFacade {
+public class MapperTopicStore {
```
Comment (petardz) [inline src/main/java/org/opensearch/securityanalytics/mapper/MapperTopicStore.java:22]: thoughts on rename?

File: src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java [hunk 2]
```diff
@@ -24,39 +23,36 @@
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
+import org.opensearch.securityanalytics.action.GetMappingsViewResponse;
+
 
 import static org.opensearch.securityanalytics.mapper.MapperUtils.*;
 
-public class MapperApplier {
+public class MapperService {
```
Comment (petardz) [inline src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java:31]: thoughts on rename?

File: src/main/java/org/opensearch/securityanalytics/action/GetMappingsViewResponse.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (aliasMappings != null && aliasMappings.size() > 0) {
+            builder.field(MapperUtils.PROPERTIES, (Map<String, ?>) aliasMappings.get(MapperUtils.PROPERTIES));
+        }
+        if (unmappedIndexFields != null && unmappedIndexFields.size() > 0) {
```
Comment (petardz) [inline src/main/java/org/opensearch/securityanalytics/action/GetMappingsViewResponse.java:101]: do we want this omitted when empty or send as empty object?

File: src/main/java/org/opensearch/securityanalytics/action/GetMappingsViewResponse.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            builder.field("unmapped_index_fields", unmappedIndexFields);
+        }
+        if (unmappedFieldAliases != null && unmappedFieldAliases.size() > 0) {
```
Comment (petardz) [inline src/main/java/org/opensearch/securityanalytics/action/GetMappingsViewResponse.java:104]: same here as above

**Review by sbcd90 — DISMISSED** [review]

**Review by sbcd90 — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Pull request merged**